src/lib/es3.d.ts

```diff
@@ -0,0 +1,3767 @@
+     */
+    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): PromiseLike<TResult1 | TResult2>;
+}
+
+/**
+ * Represents the completion of an asynchronous operation
+ */
+interface Promise<T> {
+    /**
+     * Attaches callbacks for the resolution and/or rejection of the Promise.
+     * @param onfulfilled The callback to execute when the Promise is resolved.
+     * @param onrejected The callback to execute when the Promise is rejected.
+     * @returns A Promise for the completion of which ever callback is executed.
+     */
+    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): Promise<TResult1 | TResult2>;
+
+    /**
+     * Attaches a callback for only the rejection of the Promise.
+     * @param onrejected The callback to execute when the Promise is rejected.
+     * @returns A Promise for the completion of the callback.
+     */
+    catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): Promise<T | TResult>;
+}
+
+interface ArrayLike<T> {
+    readonly length: number;
+    readonly [n: number]: T;
+}
+
+/**
+ * Make all properties in T optional
+ */
+type Partial<T> = {
+    [P in keyof T]?: T[P];
+};
+
+/**
+ * Make all properties in T readonly
+ */
+type Readonly<T> = {
+    readonly [P in keyof T]: T[P];
+};
+
+/**
+ * From T pick a set of properties K
+ */
+type Pick<T, K extends keyof T> = {
+    [P in K]: T[P];
+};
+
+/**
+ * Construct a type with a set of properties K of type T
+ */
+type Record<K extends string, T> = {
+    [P in K]: T;
+};
+
+/**
+ * Marker for contextual 'this' type
+ */
+interface ThisType<T> { }
+
+/**
+ * Represents a raw buffer of binary data, which is used to store data for the
+ * different typed arrays. ArrayBuffers cannot be read from or written to directly,
+ * but can be passed to a typed array or DataView Object to interpret the raw
+ * buffer as needed.
+ */
+interface ArrayBuffer {
+    /**
+     * Read-only. The length of the ArrayBuffer (in bytes).
+     */
+    readonly byteLength: number;
+
+    /**
+     * Returns a section of an ArrayBuffer.
+     */
+    slice(begin: number, end?: number): ArrayBuffer;
+}
+
+/**
+ * Allowed ArrayBuffer types for the buffer of an ArrayBufferView and related Typed Arrays.
+ */
+interface ArrayBufferTypes {
+    ArrayBuffer: ArrayBuffer;
+}
+type ArrayBufferLike = ArrayBufferTypes[keyof ArrayBufferTypes];
+
+interface ArrayBufferConstructor {
+    readonly prototype: ArrayBuffer;
+    new (byteLength: number): ArrayBuffer;
+    isView(arg: any): arg is ArrayBufferView;
+}
+declare const ArrayBuffer: ArrayBufferConstructor;

```

Comment:
@rbuckton That would certainly be a clean approach! To where would you suggest moving the [`Decorator` types](https://github.com/SaschaNaz/TypeScript/blob/48d2253ee2c5c55a78b4c4a004a1557fd2689976/src/lib/es3.d.ts#L943-L955)? As far as JavaScript is concerned, these are still a [proposal](https://github.com/tc39/proposal-decorators) (perhaps even _more_ of a reason to get them out of the EC3/EC5 libraries. Given that even in [TypeScript](https://www.typescriptlang.org/docs/handbook/decorators.html) they require the `experimentalDecorators` compiler option, perhaps a library like the following would work:

* _lib.experimental.decorator.d.ts_ - Decorator [type information](https://github.com/SaschaNaz/TypeScript/blob/48d2253ee2c5c55a78b4c4a004a1557fd2689976/src/lib/es3.d.ts#L943-L955).

These could probably exist on their own, to be honest. They were initially brought into `core` before it became `es5.d.ts` and seem to have just stuck around. There appear to be no interdependencies with other types in either `lib.d.ts` _or_ `lib.dom.d.ts`. They _are_ used in TypeScript when the `experimentalDecorators` option is set, but perhaps that setting can also imply `lib=["experimental.decorator"]`?